generator/config/aggregation-operators/firstN.yaml

```diff
@@ -0,0 +1,23 @@
+# $schema: ../schema.json
+name: $firstN
```

Comment:
I have a name conflict with `$firstN` that is both [aggregation accumulator](https://www.mongodb.com/docs/manual/reference/operator/aggregation/firstN/) and [array operator](https://www.mongodb.com/docs/manual/reference/operator/aggregation/firstN-array-element/). Same for `$lastN`.

Reply:
Both of these fall under aggregation, correct? And not the query namespace.

The API for both of these operators looks similar, but I assume they are distinct and there's nothing to say other operators and accumulators with the same name might differ more significantly. In this case, would it make sense to organize accumulators under their own namespace?

I actually notice that there are two groupings for accumulators in the docs:

 * [Accumulators ($group, $bucket, $bucketAuto, $setWindowFields)](https://www.mongodb.com/docs/manual/reference/operator/aggregation/#accumulators---group---bucket---bucketauto---setwindowfields-)
 * [Accumulators (in Other Stages)(https://www.mongodb.com/docs/manual/reference/operator/aggregation/#accumulators--in-other-stages-)

It's hard to predict how we might encounter naming conflicts in the future, but do you think it'd be worth organizing the many aggregation operators by separate namespaces to group them logically?

We could also do the same for [Query and Projection](https://www.mongodb.com/docs/manual/reference/operator/query/#query-and-projection-operators) operators, which have far fewer categories, if you like.

Reply:
If we over-categorize, it could be confusing for developers, who will have to guess which category we've put the operator they want to use. 
Also, I don't want to duplicate Yaml files for operator what would be in multiple categories, like `$avg` or `$count`.

Reply:
Is this file modeling the array operator or accumulator? The `link` and `description` fields below suggest the array operator , but I see `type: [ Accumulator ]` as well.

Reply:
> Is this file modeling the array operator or accumulator? The `link` and `description` fields below suggest the array operator , but I see `type: [ Accumulator ]` as well.

I can see how over-categorization could be problematic (assuming users need to reference class names and don't use factory methods). What is your proposed solution?